framework/source/class/qx/dev/unit/TestFunction.js

```diff
@@ -109,16 +109,30 @@ qx.Class.define("qx.dev.unit.TestFunction",
       var inst = this.getTestClass();
       var method = this.getName();
 
-      inst.setTestFunc(this);
-      inst.setTestResult(testResult);
+      inst.set({
+        testFunc:this,
+        testResult: testResult
+      });
 
-      testResult.run(this, function()
-      {
+      testResult.run(this, function() {
+        var ret;
         try {
-          inst[method]();
+          ret = inst[method]();
         } catch (ex) {
           throw ex;
         }
+        if (ret && ret.constructor && ret.constructor.name === "Promise") {
+          ret
+          .then(
+            ()=>inst.resume()
```

Comment:
It seems that the linter job in travis fails on the arrow function syntax.
It may be odd, but could you please change this to explicit function declarations?
Or should the linter be adapted to accept ES6?

Reply:
fixed